### PR TITLE
[CLI-820]: Fix Nesting check for runs using defined_metric

### PR DESCRIPTION
**Pull request by KyleGoyette**
https://wandb.atlassian.net/browse/CLI-820

Description
-----------

This PR modfiies the check for nested metrics in handler.py to handle metrics that are logged as dicts (like wandb.Tables) so they are no longer stripped from summary updates.

Testing
-------

Tested here: https://wandb.ai/kylegoyette/def_met_cc/runs/182gdfdi?workspace=user-kylegoyette

### CODE REVIEW

**Review by raubitsj — CHANGES_REQUESTED** [review]
couple changes requested:
- lets not use wandb.utils,  maybe somewhere in lib
- consider how data_types.py will need to be in sync with metric_is_wandb_dict()

File: wandb/data_types.py [hunk 1]
```diff
@@ -68,6 +68,7 @@
     )
 
 __all__ = [
+    "Audio",
```
Comment (KyleGoyette) [inline wandb/data_types.py:71]: I'm not sure if adding this here is okay?

Comment (tssweeney) [reply wandb/data_types.py:71]: Nice catch, yes, this should be here

File: wandb/sdk/lib/handler_util.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import wandb.data_types as data_types
+
+
+def get_types():
```
Comment (tssweeney) [inline wandb/sdk/lib/handler_util.py:4]: prints should be removed

**Review by tssweeney — CHANGES_REQUESTED** [review]
The different "types" / "type systems" are growing, and this `log_type` seems to increase that surface area. Currently, it seems like `artifact_type` is what you are looking for - this is just the string identifier of the media type's json payload. If I am am not mistaken, there are three cases:
- current `artifact_type` matches what you would have for `log_type`
- current `artifact_type` is missing
- current `artifact_type` does not match (the only case here is "table" since it is stored differently in runs & artifacts).

I am curious if we can just use `artifact_type` for these other classes and just have a special case for `table` in the helper method? These other classes will eventually be ported to artifacts, so it might make sense to just use that?

On L459 of wandb_artifacts.py we have a allow-list of classes which we currently support in artifacts. You can safely add `artifact_type` to the missing classes without the risk of messing with artifacts since that list controls explicitly what can be added.

Oh, and if you think combining them makes sense, i could see changing `artifact_type` to `log_type` if that is what you prefer since the name is a bit more appropriate (:

**Comment by KyleGoyette** [comment]
Thanks @tssweeney for the detailed breakdown. Good to know I can add in the artifact_types to those missing it. If we agree that log_type makes more sense I can move in that direction. I'm fine with adding a special case for tables, so long as @raubitsj is all right with it.

**Comment by raubitsj** [comment]
> Thanks @tssweeney for the detailed breakdown. Good to know I can add in the artifact_types to those missing it. If we agree that log_type makes more sense I can move in that direction. I'm fine with adding a special case for tables, so long as @raubitsj is all right with it.

@KyleGoyette I am fine with what you decide to do.  either doing the minimal changes in this PR (remove the print, use private _log_type, or consolidating with `_log_type` given that I know you want this fix out in the upcoming release.

**Review by raubitsj — APPROVED** [review]

**Pull request merged**